docs/develop/Architecture.md

```diff
@@ -0,0 +1,77 @@
+- [ElectionGuard Python API][election-guard-web-api-source]
+- [ElectionGuard React UI][election-guard-ui-source]
+
+The specification repository focuses on explaining ElectionGuard as a whole including documentation and the specification itself. The other repositories are all reference implementations. These reference implementations are designed to demonstrate to developers how to implement a system using the ElectionGuard specification. In depth descriptions of the repositories are detailed in the [repositories][repositories] page.
+
+With the specification as guidance, the reference repositories take two pathways; one starting with Python and one starting with C++. Each of these creating a stack. This chart shows the relationship of the specification to the reference implementations by repository. This paints the picture of how the repositories are related to each other.  
+
+```mermaid
+flowchart TD
+    spec["electionguard"] -.-> |Python|e-python["electionguard-python"];
+    spec["electionguard"] -.-> |C++|e-cpp["electionguard-cpp"];
+    e-python --> web-api["electionguard-api-python"];
+    web-api --> e-ui["electionguard-ui"];
+```
+
+Using more plain language, the specification is the guide used to create the core repositories. There are two main paths within the core reference implementations of ElectionGuard. At the base, one is written in Python and one is written in C++. These can used together, but have different feature sets.  
+
+```mermaid
+flowchart TD
+    specification -.-> |Python|python["full featured core"];
+    specification -.-> |C++|cpp["encryption core"];
+    python --> api["web-api"];
+    api --> ui["user interface"];
+```
+
+The Python reference implementation is full featured including all elements of the specification to showcase ElectionGuard to its fullest. The C++ reference implementation is designed with a focus on encrypting ballots. Its goal is to allow developers to use the ElectionGuard encryption on encryption devices.
+
+| Feature           | `electionguard-python` |  `electionguard-cpp` |
+| ----------------- | :--------------------: | :------------------: |
+| Ballot Encryption | :material-check:       | :material-check:     |
+| Ballot Decryption | :material-check:       |                      |
+| Key Generation    | :material-check:       |                      |
+| Key Ceremony      | :material-check:       |                      |
+| Tally Ceremony    | :material-check:       |                      |
+| Tally Decryption  | :material-check:       |                      |
+
+
+## Python Stack
+
+The Python reference implementation is full featured including all elements of the specification to showcase ElectionGuard to its fullest. There is a full stack provided for Python so there are many different approaches and entry points a developer could use when starting here. The following suggested approaches can be used to develop apps like those listed in the [applications][application-paths] list.
```

Comment:
There is 1 python implementation of the specification.  The second one is the api reference implementation.

Reply:
@SteveMaier-IRT There seems to be some serious problems with terminology here, and I get the impression you and @keithrfung are using the term reference implementation to mean different things.  For Keith there are very clearly two Python reference implementations.  Then confusing things further you're now introducing two new concepts "implementation of the specification" and "api reference implementation".  I like the distinction you've made, and I suspect it's helpful, but there's nothing in the documentation I've seen thus far that draws those distinctions.  I'm beginning to feel like it would be productive to have a voice conversation about all this.